lib/ddtrace/span/external_resource.rb

```diff
@@ -0,0 +1,38 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  class Span
```

Comment:
Generally not a fan of re-opening class for edits like this, even if the principle purpose is to namespace `ExternalResource`. While I do appreciate the idea of keeping the namespacing of this module closely coupled with `Span` (since it exists solely to extend that class), I'm hoping we can find a better way to do this without nesting constants within `class Span` that are not in `span.rb`. This implementation would be acceptable if `ExternalResource` remained within `span.rb`, but if the module should move to another file, then I believe the namespacing should change to reflect that without re-opening classes for edits.

For `Analytics` and `ForcedTracing` we nested the modules in those namespaces, so that's one pattern we could follow for consistency. I can't say I love that either, so if there's another way we can split this module into its own file without re-opening a class, then I'd be happy to entertain that as well.